includes/tuque_wrapper.inc

```diff
@@ -233,6 +233,52 @@ class IslandoraFedoraObject extends FedoraObject {
       throw $e;
     }
   }
+
+  /**
+   * Purge a datastream.
+   *
+   * Invokes datastream altered/purged hooks and calls the API-M method.
+   *
+   * @see FedoraObject::purgeObject()
+   */
+  public function purgeDatastream($id) {
+    $this->populateDatastreams();
+
+    if (!array_key_exists($id, $this->datastreams)) {
+      return FALSE;
+    }
+    $context = array(
+      'action' => 'purge',
+      'purge' => TRUE,
+      'delete' => FALSE,
+      'block' => FALSE,
+    );
+    try {
+      islandora_alter_datastream($this, $this[$id], $context);
+      $action = $context['block'] ? 'block' : FALSE;
+      $action = (!$action && $context['delete']) ? 'delete' : $action;
+      $action = !$action ? 'purge' : $action;
+      switch ($action) {
+        case 'block':
+          throw new Exception('Purge Datastream was blocked.');
+
+        case 'delete':
+          $this[$id]->state = 'D';
+          return array();
+
+        default:
+          return parent::purgeDatastream($id);
```

Comment:
Might've gone too far here... I was just expecting this to just become something like:
```php
$to_return = parent::purgeDatastream($id);
islandora_invoke_datastream_hooks(ISLANDORA_DATASTREAM_PURGED_HOOK, $object->models, $dsid, $object, $dsid);
return $to_return;
```

instead of breaking out separately to the API-M class.